controllers/chat/group/leaveGroupMember.js

```diff
@@ -0,0 +1,34 @@
+const BetterSocialCore = require('../../../services/bettersocial');
+const {ResponseSuccess} = require('../../../utils/Responses');
+const ErrorResponse = require('../../../utils/response/ErrorResponse');
+const Getstream = require('../../../vendor/getstream');
+
+const leaveGroupMember = async (req, res) => {
+  const {userId} = req;
+  const {channelId} = req.body;
+
+  try {
+    const response = await Getstream.chat.removeGroupMember(channelId, userId);
+    const {channel, channelApiResponse} = response.data || {};
+
+    let channelResponse;
+    try {
+      const {newChannelName, betterChannelMember, betterChannelMemberObject, updatedChannel} =
+        await BetterSocialCore.chat.updateBetterChannelMembers(channel, channelApiResponse, true);
+
+      channelResponse = updatedChannel;
+
+      channelApiResponse.channel.name = newChannelName;
+      channelApiResponse.channel.better_channel_member = betterChannelMember;
+      channelApiResponse.channel.better_channel_member_object = betterChannelMemberObject;
+    } catch (e) {
+      console.log(e);
+    }
+
+    return ResponseSuccess(res, response?.message, 200, channelResponse);
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

The use of optional chaining (`response?.message`) suggests that `response` might be `undefined`. However, if `response` is `undefined`, it would cause an error at line 12 where you're trying to destructure `data` from `response`. You should ensure `response` is defined before accessing its properties.

```diff
-     return ResponseSuccess(res, response?.message, 200, channelResponse);
+     return ResponseSuccess(res, response.message, 200, channelResponse);
```


<!-- This is an auto-generated comment by OSS CodeRabbit -->